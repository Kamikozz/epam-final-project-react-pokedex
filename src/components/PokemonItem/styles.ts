import { createStyles } from "@material-ui/core";
import { Theme } from "@material-ui/core/styles";

const styles = (theme: Theme) => createStyles({
  badgeWrapper: {
    display: "block",
    overflow: "hidden",
    borderRadius: "4px"
  },
  badge: {
    margin: "17px",
    padding: "25px",
    width: 0,
    fontSize: "20px",
    borderRadius: "0 50%",
    backgroundColor: "red"
  },
  card: {
    textAlign: "center",
    color: theme.palette.text.secondary
  },
  media: {
    height: 300,
    backgroundSize: "contain"
  },
  pokemonName: {
    textTransform: "uppercase"
  }
});

export default styles;
