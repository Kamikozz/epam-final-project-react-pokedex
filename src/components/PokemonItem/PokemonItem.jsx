import React from "react";
import { Link } from "react-router-dom";

import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import Badge from "@material-ui/core/Badge";

const styles = theme => ({
  badgeWrapper: {
    display: "block",
    overflow: "hidden",
    borderRadius: "4px"
  },
  badge: {
    margin: "17px",
    padding: "25px",
    width: 0,
    fontSize: "20px",
    borderRadius: "0 50%",
    backgroundColor: "red"
  },
  card: {
    textAlign: "center",
    color: theme.palette.text.secondary
  },
  media: {
    height: 300,
    backgroundSize: "contain"
  },
  pokemonName: {
    textTransform: "uppercase"
  }
});

class PokemonItem extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      classes,
      key,
      pokemonId,
      name,
      date,
      cardActions,
      link
    } = this.props;
    const pokemonImage = require(`../../assets/pokemons/${pokemonId}.png`);
    return (
      <Grid key={key} item xs={12} sm={6} md={4} lg={3}>
        <Badge
          className={classes.badgeWrapper}
          classes={{ badge: classes.badge }}
          color="primary"
          badgeContent={pokemonId}
          max={99999} // max pokemonId found in db.json
        >
          <Card className={classes.card}>
            <CardActionArea
              {...(link
                ? {
                    component: Link,
                    to: `/pokemons/${pokemonId}`
                  }
                : {})}
            >
              <CardMedia
                className={classes.media}
                image={pokemonImage}
                title={name}
              />
              <CardContent>
                <Typography
                  className={classes.pokemonName}
                  variant="h4"
                  component="h4"
                >
                  {name}
                </Typography>
                {date && (
                  <Typography component="p">{`Caught at: ${date}`}</Typography>
                )}
                {/*<Typography component="p">*/}
                {/*{this.getPokemonDescription(pokemon.name)}*/}
                {/*</Typography>*/}
              </CardContent>
            </CardActionArea>
            {cardActions}
          </Card>
        </Badge>
      </Grid>
    );
  }
}

export default withStyles(styles)(PokemonItem);
