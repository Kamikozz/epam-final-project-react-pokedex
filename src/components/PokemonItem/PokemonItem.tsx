import React from "react";
import { Link } from "react-router-dom";

import { WithStyles } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";
import Badge from "@material-ui/core/Badge";

import styles from "./styles";

interface Props extends WithStyles<typeof styles> {
  classes: {
    badgeWrapper: string;
    badge: string;
    card: string;
    media: string;
    pokemonName: string;
  };
  key: number;
  pokemonId: number;
  name: string;
  date?: string;
  cardActions?: JSX.Element | boolean;
  link?: boolean;
};

const LinkToPokemonsPage = (props: any) => <Link {...props} />;

const PokemonItem = (props: Props) => {
  const {
    classes,
    key,
    pokemonId,
    name,
    date,
    cardActions,
    link
  } = props;
  const pokemonImage = require(`../../assets/pokemons/${pokemonId}.png`);
  const cardActionAreaProps = link ? {
    component: LinkToPokemonsPage,
    to: `/pokemons/${pokemonId}`
  } : {};
  return (
    <Grid key={key} item xs={12} sm={6} md={4} lg={3}>
      <Badge
        className={classes.badgeWrapper}
        classes={{ badge: classes.badge }}
        color="primary"
        badgeContent={pokemonId}
        max={99999} // max pokemonId found in db.json
      >
        <Card className={classes.card}>
          <CardActionArea
            {...cardActionAreaProps}
          >
            <CardMedia
              className={classes.media}
              image={pokemonImage}
              title={name}
            />
            <CardContent>
              <Typography
                className={classes.pokemonName}
                variant="h4"
                component="h4"
              >
                {name}
              </Typography>
              {date && (
                <Typography component="p">{`Caught at: ${date}`}</Typography>
              )}
              {/*<Typography component="p">*/}
              {/*{this.getPokemonDescription(pokemon.name)}*/}
              {/*</Typography>*/}
            </CardContent>
          </CardActionArea>
          {cardActions}
        </Card>
      </Badge>
    </Grid>
  );
};

export default withStyles(styles)(PokemonItem);
