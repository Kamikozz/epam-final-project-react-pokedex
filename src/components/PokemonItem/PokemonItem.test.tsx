import React from 'react';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { render } from '@testing-library/react';

import PokemonItem from './PokemonItem';

it('renders without crashing', () => {
  render(
    <Router>
      <PokemonItem pokemonId={1} name={'some name'} />
    </Router>
  );
});

it('matches snapshot', () => {
  const { container } = render(
    <Router>
      <PokemonItem pokemonId={1} name={'some name'} />
    </Router>
  );
  expect(container.firstChild).toMatchSnapshot()
});

