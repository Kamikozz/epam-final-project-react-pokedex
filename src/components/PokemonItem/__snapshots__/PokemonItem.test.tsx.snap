// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="MuiGrid-item-196 MuiGrid-grid-xs-12-235 MuiGrid-grid-sm-6-243 MuiGrid-grid-md-4-255 MuiGrid-grid-lg-3-268"
>
  <span
    class="MuiBadge-root-292 PokemonItem-badgeWrapper-190"
  >
    <div
      class="MuiPaper-root-300 MuiPaper-elevation1-303 MuiPaper-rounded-301 MuiCard-root-299 PokemonItem-card-192"
    >
      <button
        class="MuiButtonBase-root-330 MuiCardActionArea-root-327"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root-333 PokemonItem-media-193"
          style="background-image: url(1.png);"
          title="some name"
        />
        <div
          class="MuiCardContent-root-335"
        >
          <h4
            class="MuiTypography-root-336 MuiTypography-h4-351 PokemonItem-pokemonName-194"
          >
            some name
          </h4>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight-329"
        />
        <span
          class="MuiTouchRipple-root-372"
        />
      </button>
    </div>
    <span
      class="MuiBadge-badge-293 PokemonItem-badge-191 MuiBadge-colorPrimary-294"
    >
      1
    </span>
  </span>
</div>
`;
